src/dmd/dtemplate.d

```diff
@@ -5095,6 +5102,12 @@ extern (C++) class TemplateParameter
 
     abstract bool hasDefaultArg();
 
+    override const(char)* toChars() { return this.ident.toChars(); }
+    override DYNCAST dyncast() const pure @nogc nothrow @safe
+    {
+        return DYNCAST.templateparameter;
```

Comment:
also all dead code. Can you write a test case for it?

Reply:
It will be used in a later PR